mmeval/metrics/lvis.py

```diff
@@ -0,0 +1,338 @@
+                prediction and ground truth of an image. This list has already
+                been synced across all ranks.
+
+        Returns:
+            dict: The computed metric. The keys are the names of
+            the metrics, and the values are corresponding results.
+        """
+        tmp_dir = None
+        if self.outfile_prefix is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            outfile_prefix = osp.join(tmp_dir.name, 'results')
+        else:
+            outfile_prefix = self.outfile_prefix
+        
+        # split gt and prediction list
+        preds, gts = zip(*results)
+
+        if self._lvis_api is None:
+            # use converted gt json file to initialize coco api
+            print('Converting ground truth to coco lvis format...')
+            coco_json_path = self.gt_to_coco_json(
+                gt_dicts=gts, outfile_prefix=outfile_prefix)
+            self._lvis_api = LVIS(coco_json_path)
+
+        # handle lazy init
+        if len(self.cat_ids) == 0:
+            self.cat_ids = self._lvis_api.get_cat_ids()
+        if len(self.img_ids) == 0:
+            self.img_ids = self._lvis_api.get_img_ids()
+
+        # convert predictions to coco format and dump to json file
+        result_files = self.results2json(preds, outfile_prefix)
+
+        eval_results: OrderedDict = OrderedDict()
+        if self.format_only:
+            print('results are saved in '
+                        f'{osp.dirname(outfile_prefix)}')
+            return eval_results
+
+        lvis_gt = self._lvis_api
+
+        for metric in self.metrics:
+            print(f'Evaluating {metric}...')
+
+            try:
+                lvis_dt = LVISResults(lvis_gt, result_files[metric])
+            except IndexError:
+                print(
+                    'The testing results of the whole dataset is empty.')
+                break
+
+            iou_type = 'bbox' if metric == 'proposal' else metric
+            lvis_eval = LVISEval(lvis_gt, lvis_dt, iou_type)
+            lvis_eval.params.imgIds = self.img_ids
+            metric_items = self.metric_items
+            if metric == 'proposal':
+                lvis_eval.params.useCats = 0
+                lvis_eval.params.maxDets = list(self.proposal_nums)
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
+                if metric_items is None:
+                    metric_items = ['AR@300', 'ARs@300', 'ARm@300', 'ARl@300']
+                for k, v in lvis_eval.get_results().items():
+                    if k in metric_items:
+                        val = float('{:.3f}'.format(float(v)))
+                        eval_results[k] = val
+
+            else:
+                lvis_eval.evaluate()
+                lvis_eval.accumulate()
+                lvis_eval.summarize()
+                lvis_results = lvis_eval.get_results()
+                if self.classwise:  # Compute per-category AP
```

Comment:
we can first calculate AP and add them into eval_results, or it will first print classwise result.